### DISCUSSION TITLE
Migrating from Cocos2d-x 4.0 with SDKBox to axmol

### DISCUSSION THREAD

**Discussion by qoozta** [axmolengine/axmol#654] [Q&A] [answered]
Hi, I'm trying to migrate from cocos2d-x 4.0, in my project I'm using SDKBox for IAP, on iOS it was success, however on android I can only build for debug variant, I got error when building release variant, I have tired to turn off proguard but didn't work.
This is the error:
`ld: error: undefined symbol: cocos2d::cocos2dVersion()
referenced by Tracking.o:(_GLOBAL__sub_I_Tracking.cpp) in archive ../../../../jni/sdkbox/libs_c++_static/arm64-v8a/libsdkbox.a`
In debug variant if use release signing config, everything works perfectly.
Any thoughts which part going wrong?

Many thanks.

**Comment by halx99** [comment, marked as answer]
This problem shoud be fixed by the latest commit https://github.com/adxeproject/adxe/commit/7dcffeeb9a794b6b86d8d739abe01e72d2b7839e

**Reply by qoozta** [reply]
Thanks for quick answering!

**Reply by aismann** [reply]
@qoozta 
What is needed to add SDKBox to axis?

**Reply by qoozta** [reply]
Since SDKBox is not open source, all we can do is try to integrate it and compile to see what cocos dependancies was used. Also it was stopped maintain in Dec 2021, I will keep using it as temporary solution until I finish the native code for those features such as IAP / Game Center etc.
I'm much appreciate axis keep the compatibility for Cocos2d-x and SDKBox during the transition process. If you need any help for compatibility tests, please let me know.
